Sources/TuistDependencies/CarthageGraphLoader.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import TSCBasic
+import TuistCore
+
+public protocol CarthageGraphLoading {
+    /// Loads the given dependencies at the given path.
+    /// - Parameter dependencies: Array of dependencies to be fetched with carthage
+    /// - Parameter path: Path to the platform directory inside Carthage Build folder.
+    /// - Returns DependencyGraph with transitive dependencies
+    func load(dependencies: [CarthageDependency], atPath path: AbsolutePath) throws -> DependencyGraph
+}
+
+public struct CarthageGraphLoader: CarthageGraphLoading {
```

Comment:
Maybe `TuistLoader` layer will be a better place it. I thinking at loud. 🤔

Reply:
Yeah good idea. I'll move it.